### DISCUSSION TITLE
Syslog Parsing Issues

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#4400] [Unsupported Versions] [answered]
Hello!

I want to do some custom syslog parsing and followed the instructions to copy a custom parsing file in /opt/so/saltstack/local/salt/elasticsearch/files/ingest and did a restart of Elasticsearch on my master and search node without an error. I tested my grok paterns in Grok Debugger without errors. But my syslogs do not get parsed (I just see the unparsed messages in Elasticsearch). Do I have to put the file on the master and the search node? Do I have to rename it (it still calls syslog)? What additional steps do I have to do? Thanks in advance!

Cheers, Ben

**Comment by ben-sec** [comment]
OK, 12 hrs later and it's working. Do pipeline changes need some time to have an effect?

**Reply by ben-sec** [reply]
Seriously, what do I have to do to instruct Elasticsearch to use the updated pipeline? It takes ages for him to realize that something has changed. Thanks!

**Comment by ben-sec** [comment]
Since I never got an answer, I give it another try (since I work on my syslog parsing and it drives me crazy ...): Can someone answer me my question above (What do I have to do to instruct Elasticsearch to use the updated pipeline?)? Thanks in advance!

**Reply by weslambert** [reply]
Sorry for the delayed response -- this must have gotten lost in the shuffle (apologies, but we don't respond/track these issues to the degree of official support). You only need to make the change on the manager.  You can check to see if your custom pipeline is loaded with something like the following:

`so-elasticsearch-pipeline-view syslog`

To see the stats for the pipeline (such as failures/successes) use:

`so-elasticsearch-pipeline-stats syslog`

Thanks,
Wes

**Reply by ben-sec** [reply]
The point is, that I have updated my current and working custom syslog pipeline in `/opt/so/saltstack/local/salt/elasticsearch/files/ingest/syslog` and I also find updated versions that reflect these changes in `/opt/so/conf/elasticsearch/ingest/syslog` and `/var/cache/salt/minion/files/base/elasticsearch/files/ingest/syslog` (all on my manager node). I did already a `salt-call state.highstate` and `so-elasticsearch-restart`, but my changes don't go live. If I run `so-elasticsearch-pipeline-view syslog` I see the old version of my pipeline without my changes (same in Kibana->Stack Management->Ingest Pipelines).

**Reply by ben-sec** [reply]
Can I try to change the ingest pipeline directly in Kibana?

![image](https://user-images.githubusercontent.com/81571847/148515552-cb876264-bc75-46e8-873f-276de12c2ee4.png)

**Reply by weslambert** [reply]
The pipeline files should be loaded assuming Salt has detected a change in the file.  As far as editing within Kibana, you can do so for testing, but I wouldn't recommend it for long-term modifications, as it may get overwritten by the pipeline file on disk since it is the authoritative source.

**Comment by ben-sec** [comment, marked as answer]
Finally I found the issue: I got two errors in my Grok patterns. The Kibana Grok-Debugger did not complain about them and they worked fine there, but I got Jason string errors when I tried to use them in the Grok filter above in Kibana.

1. `\(` worked in the Kibana Grok Debugger, but not in the Ingest Pipeline. `\\(` did the job.
2.  `\s*` worked in the Kibana Grok Debugger for a whitespace, but not in the Ingest Pipeline. A normal space " " works.

When I fixed these errors, SO passed my configuration successfully from `/opt/so/saltstack/local/salt/elasticsearch/files/ingest/syslog` to the Elasticsearch pipeline (after a `salt-call state.highstate` and `so-elasticsearch-restart` on my manager and search node.
